### PR TITLE
PT github-actions 

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Add Github Action for PyTorch precommit tests
Add mark `cuda` to separate tests on cpu and cuda 

### Reason for changes

Migration precommit tests to GA 

### Related tickets

133008

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -145,6 +145,12 @@ install-models-hub-torch:
 test-torch:
 	pytest ${COVERAGE_ARGS} tests/torch -m "not weekly and not nightly and not models_hub" --junitxml ${JUNITXML_PATH} $(DATA_ARG)
 
+test-torch-cpu:
+	pytest ${COVERAGE_ARGS} tests/torch -rS -m "not cuda and not weekly and not nightly and not models_hub" --junitxml ${JUNITXML_PATH} $(DATA_ARG)
```
Comment (mryzhov) [inline Makefile:149]: let's try run them in parallel
```suggestion
	pytest -n logical ${COVERAGE_ARGS} tests/torch -rS -m "not cuda and not weekly and not nightly and not models_hub" --junitxml ${JUNITXML_PATH} $(DATA_ARG)
```

**Review by daniil-lyakhov — APPROVED** [review]
Test parameters looks ok 👍

File: .github/workflows/precommit.yml [hunk 2]
```diff
@@ -89,3 +89,73 @@ jobs:
...
+        run: make install-torch-test
+      - name: Run PyTorch precommit test scope
+        run: |
+          make test-torch-cpu
+  pytorch-cuda:
+    timeout-minutes: 40
+    defaults:
+      run:
+        shell: bash
+    runs-on: aks-linux-4-cores-28gb-gpu-testla-t4
```
Comment (KodiaqQ) [inline .github/workflows/precommit.yml:126]: Tes**t**la or tesla?

File: .github/workflows/precommit.yml [hunk 3]
```diff
@@ -89,3 +89,73 @@ jobs:
           token: ${{ secrets.CODECOV_TOKEN }}
           name: coverage_openvino
           flags: OPENVINO
+  pytorch-cpu:
+    timeout-minutes: 40
```
Comment (KodiaqQ) [inline .github/workflows/precommit.yml:93]: Are the 40 minutes enough for cpu?

File: .github/workflows/precommit.yml [hunk 4]
```diff
@@ -89,3 +89,73 @@ jobs:
...
+          sudo sh cuda_12.1.1_530.30.02_linux.run --toolkit --silent
+      - name: Runner info
+        continue-on-error: true
+        run: |
+          export PATH=/usr/local/cuda-12.1/bin${PATH:+:${PATH}}
+          export LD_LIBRARY_PATH=/usr/local/cuda-12.1/lib64${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
+          nvidia-smi
+          cat /proc/cpuinfo
+          nvcc --version
+      - uses: actions/checkout@v3
```
Comment (KodiaqQ) [inline .github/workflows/precommit.yml:146]: Don't we need to bump to v4? And bump all other actions to actual versions?

File: tests/torch/requirements.txt [hunk 5]
```diff
@@ -16,3 +16,4 @@ datasets~=2.14.0
 evaluate==0.3.0
 timm==0.9.2
 openvino-dev==2024.0
+pytest-xdist==3.5.0
```
Comment (alexsu52) [inline tests/torch/requirements.txt:19]: Don't forget to add "pytest-xdist" to the constraints file https://github.com/openvinotoolkit/nncf/pull/2581

**Review by alexsu52 — COMMENTED** [review]
Please add common GA for the common tests to have ability to remove the jenkins build.

File: .github/workflows/precommit.yml (hunk 2 above)
Comment (AlexanderDokuchaev) [reply .github/workflows/precommit.yml:126]: I did not choose the name 😄

File: .github/workflows/precommit.yml (hunk 3 above)
Comment (AlexanderDokuchaev) [reply .github/workflows/precommit.yml:93]: yes

File: .github/workflows/precommit.yml (hunk 4 above)
Comment (AlexanderDokuchaev) [reply .github/workflows/precommit.yml:146]: May be it's need, but in another PR.

File: tests/torch/requirements.txt (hunk 5 above)
Comment (AlexanderDokuchaev) [reply tests/torch/requirements.txt:19]: done

File: .github/workflows/precommit.yml (hunk 2 above)
Comment (AlexanderDokuchaev) [reply .github/workflows/precommit.yml]: fixed

**Comment by AlexanderDokuchaev** [comment]
@alexsu52 Common tests have already been added

https://github.com/openvinotoolkit/nncf/blob/develop/.github/workflows/precommit.yml#L14

**Review by alexsu52 — APPROVED** [review]

**Review by KodiaqQ — APPROVED** [review]

**Pull request merged**